src/main/scala/com/scality/clueso/query/MetadataQueryExecutor.scala

```diff
@@ -0,0 +1,255 @@
+    var landingTable = spark.table("landing")
+      .where(col("bucket").eqNullSafe(bucketName))
+      .select(cols: _*)
+
+    landingTable
+  }
+
+  // returns data frame together with wasCached boolean
+  def setupDf(spark : SparkSession, config : CluesoConfig, bucketName : String) : Dataset[Row]= {
+    val currentWorkers = Math.max(currentActiveExecutors(spark.sparkContext).count(_ => true), 1)
+
+    logger.info(s"Calculating DFs for bucket $bucketName – current workers = $currentWorkers")
+
+    // read df
+    var stagingTable = getColdStagingTable(spark, config, bucketName)
+    var landingTable = getColdLandingTable(spark, config, bucketName)
+
+    val colsLanding = landingTable.columns.toSet
+    val colsStaging = stagingTable.columns.toSet
+    val unionCols = colsLanding ++ colsStaging
+
+    // window function over union of partitions bucketName=<specified bucketName>
+    val windowSpec = Window.partitionBy("key").orderBy(col("kafkaTimestamp").desc)
+
+    import SparkUtils.fillNonExistingColumns
+
+    var result = landingTable.select(fillNonExistingColumns(colsLanding, unionCols):_*)
+      .union(stagingTable.select(fillNonExistingColumns(colsStaging, unionCols):_*))
+      .withColumn("rank", row_number().over(windowSpec))
+      .where(col("rank").lt(2).and(col("type") =!= "delete"))
+      .select(col("bucket"),
+        col("kafkaTimestamp"),
+        col("key"),
+        col("type"),
+        col("message.`userMd`").as("userMd"),
+        col("message.`incrementer`").as("incrementer"),
+        col("message.`sourceFileName`").as("sourceFileName"),
+        col("message.`acl`").as("acl"),
+        col("message.`location`").as("location"),
+        col("message.`tags`").as("tags"),
+        col("message.`replicationInfo`").as("replicationInfo"),
+        col("message.`md-model-version`").as("md-model-version"),
+        col("message.`owner-display-name`").as("owner-display-name"),
+        col("message.`owner-id`").as("owner-id"),
+        col("message.`content-length`").as("content-length"),
+        col("message.`content-type`").as("content-type"),
+        col("message.`last-modified`").as("last-modified"),
+        col("message.`content-md5`").as("content-md5"),
+        col("message.`x-amz-server-version-id`").as("x-amz-server-version-id"),
+        col("message.`x-amz-storage-class`").as("x-amz-storage-class"),
+        col("message.`x-amz-server-side-encryption`").as("x-amz-server-side-encryption"),
+        col("message.`x-amz-server-side-encryption-aws-kms-key-id`").as("x-amz-server-side-encryption-aws-kms-key-id"),
+        col("message.`x-amz-server-side-encryption-customer-algorithm`").as("x-amz-server-side-encryption-customer-algorithm"),
+        col("message.`x-amz-website-redirect-location`").as("x-amz-website-redirect-location"),
+        col("message.`isDeleteMarker`").as("isDeleteMarker"),
+        col("message.`x-amz-version-id`").as("x-amz-version-id"))
```

Comment:
I do. This is the setupDf which doesn't return any kind of result, but the maximum info before the search query criteria (where statement) is applied. Therefore we need to be sure all the things that should be searchable are in this select. Having said that, there's probably a nicer way instead of declaring all those fields